File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_set<int>a;
        for(int i=0;i<nums.size();i++)
        {
            if(a.find(nums[i])!=a.end())
            {
                return nums[i];
                break;
            }        
            a.insert(nums[i]);        }
        return -1;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        d={}
        for i in nums:
            if i not in d:
                d[i]=1
            else:
                d[i]+=1
        for i in d.keys(): 
            if d[i] > 1: 
                return i